.github/workflows/build-docker.yml

```diff
@@ -93,15 +93,16 @@ jobs:
       - name: Build and Deploy Docker image
         uses: docker/build-push-action@v2
         env:
-          IMAGE_TAG: ${{ env.IMAGE_NAME }}:${{ matrix.tag }}${{ (matrix.tag != '' && '-') || '' }}latest
-          VERSIONED_IMAGE_TAG: |
+          IMAGE_TAG: |- # If it's a release, add the version, otherwise add the `latest`
             ${{ (
               github.event.release.tag_name != '' && (
                 matrix.tag != '' && (
                   format('{0}:{1}-{2}', env.IMAGE_NAME, matrix.tag, github.event.release.tag_name)
                 ) || format('{0}:{1}', env.IMAGE_NAME, github.event.release.tag_name)
               )
-            ) || '' }}
+            ) || (
+              format('{0}:{1}{2}latest', env.IMAGE_NAME, matrix.tag, (matrix.tag != '' && '-') || ''
+            ) }}
```

Comment:
ちょっと混乱しそうだったので、Python で

<details>
<summary>Python による同等のコード</summary>
<div>

```python
'''image_tag.py'''

def get_image_tag(image_name, matrix_tag, release_tag_name):
    return release_tag_name != '' and (
        matrix_tag != '' and (
            '{0}:{1}-{2}'.format(image_name, matrix_tag, release_tag_name)
            ) or '{0}:{1}'.format(image_name, release_tag_name)
        ) or (
            '{0}:{1}{2}latest'.format(image_name, matrix_tag, (matrix_tag != '' and '-') or '')
        )

build_cases = [
    {
        'case': 'release',
        'image_name': 'UserName/voicevox_engine',
        'matrix_tag': 'cpu-ubuntu20.04',
        'release_tag_name': '0.10.preview.4',
    },
    {
        'case': 'release (empty matrix.tag)',
        'image_name': 'UserName/voicevox_engine',
        'matrix_tag': '',
        'release_tag_name': '0.10.preview.4',
    },
    {
        'case': 'master push',
        'image_name': 'UserName/voicevox_engine',
        'matrix_tag': 'cpu-ubuntu20.04',
        'release_tag_name': '',
    },
    {
        'case': 'master push (empty matrix.tag)',
        'image_name': 'UserName/voicevox_engine',
        'matrix_tag': '',
        'release_tag_name': '',
    },
]

for build_case in build_cases:
    print('case "{0}":\n\t"{1}"'.format(
        build_case['case'],
        get_image_tag(
            build_case['image_name'],
            build_case['matrix_tag'],
            build_case['release_tag_name']
        )
    ))
```
</div>
</details>

結果は以下のようになり、問題なさそうでした：

```
$ python image_tag.py
case "release":
	"UserName/voicevox_engine:cpu-ubuntu20.04-0.10.preview.4"
case "release (empty matrix.tag)":
	"UserName/voicevox_engine:0.10.preview.4"
case "master push":
	"UserName/voicevox_engine:cpu-ubuntu20.04-latest"
case "master push (empty matrix.tag)":
	"UserName/voicevox_engine:latest"
```

結果を眺めていて、もしかしたらもっと簡単に書けるかもしれないと思いました。`env.VOICEVOX_ENGINE_VERSION` が

https://github.com/VOICEVOX/voicevox_engine/blob/ba454b4d868d866d4fca97b69b0a887f1e8696ff/.github/workflows/build-docker.yml#L16-L17

のように「リリースタグがないときに latest になる」と設定されているので、次のように書けそうです：

```yaml
          IMAGE_TAG: |- # If it's a release, add the version, otherwise add the `latest`
            ${{ (
              matrix.tag != '' && (
                format('{0}:{1}-{2}', env.IMAGE_NAME, matrix.tag, env.VOICEVOX_ENGINE_VERSION)
              ) || format('{0}:{1}', env.IMAGE_NAME, env.VOICEVOX_ENGINE_VERSION)
            ) }}
```

あるいは：

```yaml
          IMAGE_TAG: |- # If it's a release, add the version, otherwise add the `latest`
            ${{ format('{0}:{1}{2}{3}', env.IMAGE_NAME, matrix.tag, (matrix.tag != '' && '-') || '', env.VOICEVOX_ENGINE_VERSION) }}
```

Reply:
検証ありがとうございます！修正しました！